Underline the word right under it

stoivo opened the pull request:
closes #164

I want to get it out there. what do you think?

stoivo commented:
And why do we only draw lines to the 20 first ones?

idank commented:
I'm not sure about the 20 lines. Want to open a new issue and add a reproduction there?

I'll look at the pull request soon, thanks.

stoivo commented:
I added a new PR for that Mark all flags #167,

Totally independent.

idank commented:
I tried reviewing the code but it's hard because you renamed variables in addition to the change you did. I'd prefer it if you didn't selectively start renaming variables because it's weird when the ones you touched are the only ones with a long name with underscores. Can you add comments instead? Thanks.

stoivo commented:
I will make some smaller easy to read commit when I have the time.

stoivo commented:
Please try to review it now, commit by commit.

idank commented on explainshell/web/static/js/es.js:
```diff
@@ -563,20 +563,21 @@ function drawgrouplines(commandselector, options) {
-        link.paths.push(new espath().addpoint(rr.left, 0).addpoint(rrright, 5));
-        link.paths.push(new espath().addpoint(rrright, 5).addpoint(rrright, 0));
-        path.addpoint(rr.left + rr.width / 2, 5); // 3
+        var command = link.option.getBoundingClientRect(),
```
commandRect?

stoivo replied:
I will do

idank commented:
Thanks, much easier. Since you're renaming some variables, let's start using camelCase.

idank commented on explainshell/web/static/js/es.js:
```diff
@@ -669,10 +673,17 @@ function drawgrouplines(commandselector, options) {
+              .addpoint(rrright, 0)
+            );
             var rrmid = d3.round(rr.left + rr.width / 2);
-            link.lines.push({x1: rrmid, y1: 6, x2: rrmid, y2: 5+unknownlinelength+command_offsett_to_canvas});
```
Any reason you changed this?

stoivo replied:
Nothing else than using the the same pattern. This is only for me to easily understand.

Revert it?

idank commented on explainshell/web/static/js/es.js:
```diff
@@ -572,6 +572,7 @@ function drawgrouplines(commandselector, options) {
             command_right = command.right - strokewidth;
 
         link.starty = command.bottom - command_wrapper.top + 1;
+        command_offsett_to_canvas = margin_between_command_and_canvas - link.starty
```
offset

stoivo replied:
oh, thanks

stoivo commented:
Hello, would you take the time to look at this?

idank commented:
I'm away from home for a while but I'll get to it when I can. Sorry for the
delay.

On Jan 19, 2017 11:19, "Simon" <notifications@github.com> wrote:

> Hello, would you take the time to look at this?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/idank/explainshell/pull/166#issuecomment-273871288>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABN-xBNmJYw4_EP6LNVEE0rKa-EVVzKjks5rT7dYgaJpZM4LJcaU>
> .
>

stoivo commented:
Yes, how are we doing?

idank commented:
I haven't forgotten. Hopefully will get to it this weekend. Thanks for your patience!

idank commented on explainshell/web/static/js/es.js:
```diff
@@ -573,10 +573,16 @@ function drawgrouplines(commandselector, options) {
             commandRight = commandRect.right - strokewidth;
 
         link.starty = commandRect.bottom - commandWrapperRect.top + 1;
+        commandOffsettToCanvas = marginBetweenCommandAndCanvas - link.starty
```
commandOffsettToCanvas -> commandOffsetToCanvas

idank replied:
also, is this missing a 'var'?

idank commented on explainshell/web/static/js/es.js:
```diff
@@ -568,15 +568,23 @@ function drawgrouplines(commandselector, options) {
-        link.paths.push(new espath().addpoint(rr.left, 0).addpoint(rrright, 5));
-        link.paths.push(new espath().addpoint(rrright, 5).addpoint(rrright, 0));
-        path.addpoint(rr.left + rr.width / 2, 5); // 3
+        var path = new espath();
```
what's going on here? is 'path' used anywhere after this?

stoivo replied:
Yes we continue to add points here
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR602
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR605
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR617
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR627
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR630
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR642

and then we add path to link
https://github.com/idank/explainshell/pull/166/files?diff=unified#diff-ab5c511b11e469baa7210dae1051b2bdR646

stoivo commented:
Nice, I fixed the typo and commented your comment.

idank commented:
Hey, I was trying this today to see how it looks and saw [this weird glitch](http://i.imgur.com/PXzcEKc.png), any idea why it's messed up like that?

idank commented:
The command I used was

    tar zcf - some-dir | ssh some-server "cd /; tar xvzf -" | tar zcf - some-dir | ssh some-server "cd /; tar xvzf -" | tar zcf - some-dir | ssh some-server "cd /; tar xvzf -"

stoivo commented:
Interesting, I'll take a look

stoivo commented:
I think this explains it 
![screen shot 2017-04-07 at 19 14 08](https://cloud.githubusercontent.com/assets/3492040/24811470/7cecc8b2-1bc6-11e7-8c7d-81ec9a13814f.png)

stoivo commented:
I need to tune 
https://github.com/stoivo/explainshell/blob/151897a9acb6d3eae0da28e13f0b20ad36d630b7/explainshell/web/static/js/es.js#L612-L617
https://github.com/stoivo/explainshell/blob/151897a9acb6d3eae0da28e13f0b20ad36d630b7/explainshell/web/static/js/es.js#L637-L642

stoivo commented:
I fixed it :tada:  I added couple of screenshots. 
![screen shot 2017-04-07 at 21 19 12](https://cloud.githubusercontent.com/assets/3492040/24816155/003ebd40-1bd8-11e7-8c36-4475cbd9bbbc.png)
![screen shot 2017-04-07 at 21 19 54](https://cloud.githubusercontent.com/assets/3492040/24816157/003f7186-1bd8-11e7-9e54-71816aa0c8cb.png)
![screen shot 2017-04-07 at 21 19 29](https://cloud.githubusercontent.com/assets/3492040/24816156/003f6614-1bd8-11e7-9438-46bb4a1a32bf.png)

I didn't know what this feature existed. Thank for feedback. You can test it again.

stoivo commented:
Ping

idank commented:
Will do my best this weekend. Thanks for your patience, it's admirable. :)

idank commented:
I'm trying to review this but honestly I can't remember what the code does and it's hard to know if anything about the existing drawing algorithm gets broken.

I think I'll pull this locally and run with it for a few days for my own personal use. If I don't see anything weird I'll take it.

Eventually I'd really like to rewrite the drawing algorithm so it supports multi line commands, is better documented and is easier to modify.

stoivo commented:
Soinds good, it was kind of weird logic

stoivo commented:
How are we doing?

idank commented:
Didn't see anything unusual. I'm away on a trip, will push it in a few days.

stoivo commented:
Seen any bugs?

idank commented:
None that I could see, so I finally pushed it! Congrats and thank you!

I'm sorry it took this long. I've been really busy with other things and I wish I had more time to work on the site, I have so many ideas in the back of my head, somewhere. But it makes me happy that people like you care about the site and continue improving it.

If you'd like, I'll be happy to send you a little token of appreciation for your contributions to the site (this isn't your first!). Reach me out privately at idan@explainshell.com and we can sort out the details.
